crates/conjure-cp-core/src/solver/mod.rs

```diff
@@ -306,10 +308,19 @@ impl<Adaptor: SolverAdaptor> Solver<Adaptor> {
     pub fn get_family(&self) -> SolverFamily {
         self.adaptor.get_family()
     }
+
+    /// Returns a reference to the adaptor for the underlying solver implementation.
+    pub fn adaptor(&self) -> &dyn SolverAdaptor {
+        self.adaptor.as_ref()
+    }
+
+    pub fn adaptor_mut(&mut self) -> &mut dyn SolverAdaptor {
+        self.adaptor.as_mut()
+    }
```

Comment:
All SolverAdaptors assume that the operations are called in a particular order, as defined by the Solver state machine, and rely on this fact for validation and potentially memory or thread safety.  Allowing the user access to the SolverAdaptor allows them to bypass the state-machine, which would likely make bad things happen. 


I don't think we should ever expose the underlying adaptor publicly; however, if it is **absolutely** necessary in some circumstances, we can discuss mitigations, e.g. using #doc[hidden], `unsafe`, or suffixing the methods names with `unchecked`....